sklearn/ensemble/_gb.py

```diff
@@ -424,23 +638,28 @@ def fit(self, X, y, sample_weight=None, monitor=None):
         X, y = self._validate_data(
             X, y, accept_sparse=["csr", "csc", "coo"], dtype=DTYPE, multi_output=True
         )
-
         sample_weight_is_none = sample_weight is None
-
         sample_weight = _check_sample_weight(sample_weight, X)
+        y = self._encode_y(y=y, sample_weight=sample_weight)
+        y = column_or_1d(y, warn=True)  # TODO: Is this still required?
 
-        y = column_or_1d(y, warn=True)
-
-        if is_classifier(self):
-            y = self._validate_y(y, sample_weight)
-        else:
-            y = self._validate_y(y)
+        self._set_max_features()
 
-        self._check_params()
+        if isinstance(self.loss, str):
+            self._loss = self._get_loss(sample_weight=sample_weight)
+        elif isinstance(self.loss, BaseLoss):
+            self._loss = self.loss
 
         if self.n_iter_no_change is not None:
             stratify = y if is_classifier(self) else None
-            X, X_val, y, y_val, sample_weight, sample_weight_val = train_test_split(
+            (
+                X_train,
```

Comment:
I do like changing these variables to `X_train`, etc. To make this PR smaller, may you open a PR that does these variable name changes?

Reply:
I did that change because `X` gets overwritten with `X_train` which I consider a bug. You're right that a separate PR might have been better.

Reply:
See #26959.